account-compression/sdk/tests/accountCompression.test.ts

```diff
@@ -0,0 +1,684 @@
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        proof
+      );
+      try {
+        await execute(provider, [replaceLeafIx], [payer]);
+        assert(false, "Replace should have failed to verify");
+      } catch { }
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+    it("Replace that leaf", async () => {
+      const previousLeaf = offChainTree.leaves[0].node;
+      const newLeaf = crypto.randomBytes(32);
+      const index = 0;
+
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        getProofOfLeaf(offChainTree, index, false, -1).map((treeNode) => {
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + MAX_DEPTH,
+        `Failed to create proof for ${MAX_DEPTH}`
+      );
+
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+
+    it("Replace that leaf with a minimal proof", async () => {
+      const previousLeaf = offChainTree.leaves[0].node;
+      const newLeaf = crypto.randomBytes(32);
+      const index = 0;
+
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        getProofOfLeaf(offChainTree, index, true, 1).map((treeNode) => {
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + 1,
+        "Failed to minimize proof to expected size of 1"
+      );
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+  });
+
+  describe("Examples tranferring appendAuthority", () => {
+    const authority = Keypair.generate();
+    const randomSigner = Keypair.generate();
+    describe("Examples transferring authority", () => {
+      it("... initializing tree ...", async () => {
```

Comment:
Maybe this should be a `beforeEach`?

Reply:
Kind of annoying to do that because we want to test multiple transactions back to back.

Reply:
You can condense `Can transfer authority` and `Attempting to replace with new authority now works` into one, since it's the check that the authority transfer actually worked, then all of the tests can be independent

Reply:
🙏 thank you for your fresh set of eyes Jon! Cleaned this up